src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -93,9 +93,10 @@ public ActivityLogEntry(AppLogLine appLog){
             message = tokens[8];
             url = tokens[9];
             if (tokens.length >= 11) {
-                if (tokens[10].contains(googleId)) {
-                    timeTaken = tokens.length == 12 ? Long.parseLong(tokens[11].trim()) : null;
+                boolean isLogId = tokens[10].contains(googleId) || tokens[10].contains("%"); 
+                if (isLogId) {
                     id = tokens[10];
+                    timeTaken = tokens.length == 12 ? Long.parseLong(tokens[11].trim()) : null;
                 } else {
                     timeTaken = Long.parseLong(tokens[10].trim());                    
                 }
```

Comment:
Extract `try` and `catch` blocks into separate methods? As a rule of thumb, try to keep constructors simple.
Similarly, see if this `if` statement can be extracted out.